docs_src/examples/Ch-OSImageWithEdgeX.md

```diff
@@ -0,0 +1,827 @@
+- configuration.toml
+- devices
+- profiles
+
+For this exercise, we will modify the default configurations and remove most default devices and resources. We will also replace the startup message set in the `configuration.toml` file.
+
+This snap should be build and uploaded to the store. We use `edgex-config-provider-example` as the snap name. Refer to [docs](https://docs.edgexfoundry.org/2.2/getting-started/Ch-GettingStartedSnapUsers/#config-provider-snap) for more details and example source code.
+
+Build:
+```
+$ snapcraft
+...
+Snapped edgex-config-provider-example_2.3_amd64.snap
+```
+
+This will build for your host architecture, so if your machine is `arm64`, it will result in a snap that has the same architecture. You can perform [remote builds](https://snapcraft.io/docs/remote-build) to build for other architectures.
+
+Let's upload the `amd64` snap and release it to the `latest/edge` channel:
+```
+snapcraft upload --release=latest/edge ./edgex-config-provider-example_2.3_amd64.snap
+```
+
+Now, we can query the snap ID from the store:
+```
+$ snap info edgex-config-provider-example | grep snap-id
+snap-id: WWPGZGi1bImphPwrRfw46aP7YMyZYl6w
+```
+We need it in the next step.
+
+
+### Add the config provider to the image
+
+We have to make three adaptations:
+
+1) Add the config provider to the model assertion
+
+```yaml title="model.yaml"
+# This snap contains our configuration files
+- name: edgex-config-provider-example
+  type: app
+  default-channel: latest/edge
+  id: WWPGZGi1bImphPwrRfw46aP7YMyZYl6w
+```
+
+Sign the model as before:
+```bash
+yq eval model.yaml -o=json | snap sign -k edgex-demo > model.signed.yaml
+```
+
+2) Remove config overrides from the gadget
+
+Commented out:
+```yaml title="gadget.yaml"
+    # # Enable app options
+    # app-options: true # not necessary because this service has it by default
+    # # Override the startup message (because we can)
+    # # The same syntax can be used to override most of the server configurations
+    # apps.device-virtual.config.service-startupmsg: "Gadget set this!"
+```
+
+!!! warning
+    It is important to do this because overrides are ineffective when configurations are replaced from a config provider.
+    This is because the config provider in our example is providing a read-only file system that doesn't allow the write access necessary to inject an environment file when setting the `app` options.
+
+3) Connect the config provider 
+
+```yaml title="gadget.yaml"
+connections:
+   -  # Connect edgex-device-virtual's plug (consumer)
+      plug: AmKuVTOfsN0uEKsyJG34M8CaMfnIqxc0:device-virtual-config
+      # to edgex-config-provider-example's slot (provider) to override the default configuration files.
+      slot: WWPGZGi1bImphPwrRfw46aP7YMyZYl6w:device-virtual-config
+```
+This internally bind-mounts provider's "res" directory on top of the consumer's "res" directory.
+
```

Comment:
You need to rebuild the pc_20-0.4_amd64.snap every time you change the `gadget.yaml` file:
```
$ snapcraft
... "--md-code-hl-number-color"
Snapped pc_20-0.4_amd64.snap
```